rskj-core/src/main/java/co/rsk/core/bc/BlockExecutor.java

```diff
@@ -401,4 +410,13 @@ public static void maintainPrecompiledContractStorageRoots(Repository track, Act
 
         return logBloom.getData();
     }
+
+    public ProgramResult getProgramResult(Keccak256 txhash) {
+        return this.transactionResults.get(txhash);
+    }
+
+    public void setRegisterProgramResults(boolean value) {
+        this.registerProgramResults = value;
+        this.transactionResults = new HashMap<>();
```

Comment:
I'd suggest to initialize `transactionResults` in the line where it's declared and here simply do `clear()`, if needed. `getProgramResult()` will crash if `transactionResults` is not initialized yet.

Reply:
Good catch, done